### ISSUE TITLE
Add support for displaying custom profile field data

### ISSUE THREAD

**Issue by timabbott** [zulip/zulip-mobile#2900]
We recently added support for custom profile fields in the Zulip server and webapp, and we should probably plan to add support for that on mobile as well.

I don't think we need to care about editing custom profile fields on mobile (it's a lot of work for not that much benefit), but we should probably have a way to see them when you get full-screen data about another user (where we show name/email/large avatar/etc.).  

`show_user_profile` in `static/js/popovers.js` has the logic for rendering these data in the webapp.

**Comment by nanohayder** [comment]
is there any update about this it will greatly help us

**Comment by gnprice** [comment]
@nanohayder I would be curious to hear more about why this would greatly help you! Either here, or a conversation might be easier on Zulip -- come say hello in `#mobile` on chat.zulip.org.

@timabbott Is there API documentation somewhere on how to interpret custom profile fields? There's an example now on https://zulipchat.com/api/get-all-users , which helps some, but the data there is pretty abstract. I imagine the table of what each numbered field means is somewhere on the `/register` response; we'll want some documentation of how that interface works in order for someone to efficiently implement this feature correctly.

**Comment by timabbott** [comment]
Nope; that's in my view the main TODO for the "users" part of the API documentation.  I expect we'll add it when we add documentation for the endpoints for custom profile fields more generally, which will at least provide other details we can reference for more details from a link.  I imagine @akashaviator will do it at some point in the next few weeks.

**Comment by timabbott** [comment]
But some brief notes:
* In this block, the `1` is the field ID, and the content is this user's value.
```
                "1": {
                    "rendered_value": "<p>+0-11-23-456-7890</p>",
                    "value": "+0-11-23-456-7890"
                },
```

* The mapping from field ID to what the field actually is (it's title, hint, and type) lives in the `custom_profile_fields` section of the `/register` response.  
* The field types are described in `custom_profile_field_types`, mainly for UI purposes.
* The `CustomProfileField` definition is the best resource of what the 7 supported field types mean.
* `get_custom_profile_field_data` in `static/js/popovers.js` is a block for parsing these that we may want to turn into shared code.

**Comment by gnprice** [comment]
Cool, thanks -- that's helpful for understanding the scope of it. Once that documentation is written, it looks like this should be a pretty straightforward task for someone to do.

Here's also the chat thread where @nanohayder discussed their use case:
https://chat.zulip.org/#narrow/stream/48-mobile/topic/Issue.20.232900/near/862453

**Comment by ray-kraesig** [comment]
For reference, the relevant (meta-)issue on the server-side is zulip/zulip#10134.

**Comment by gnprice** [comment]
> [zulip/zulip#10134](https://github.com/zulip/zulip/issues/10134)

That looks to be about a specific additional feature, which isn't complete.

Custom profile fields in general are this feature:
https://zulipchat.com/help/add-custom-profile-fields

**Comment by carlinmack** [comment]
This would greatly help our organisation where we have a custom field for pronouns (necessary for referring to people). Also, showing emails on mobile would be very helpful as we have many members with the same name and sadly they haven't set a picture.

**Comment by chrisbobbe** [comment]
See #5363 for giving privileged placement to certain "selected" fields.

**Comment by gnprice** [comment]
[Chat thread](https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1382978) about the remaining open questions in this API.
